opencl/gaussian/gaussianElim.cpp

```diff
@@ -172,6 +193,28 @@ int main(int argc, char *argv[]) {
         PrintAry(finalVec,size);
     }
 
+    //compare finalvec is enough i think
+    if (compare) // ensure precomputed result is available
+    {
+        float *compareVec = (float *) malloc(size * sizeof(float));
+        FILE *fpr = fopen(resname, "r");
+        if (fpr == NULL) {
+            printf("Error: unable to open file %s\n", resname);
+            exit(1);
+        }
+        InitAry(fpr, compareVec, size);
+        fclose(fpr);
+
+        int i;
+        for (i = 0; i < size; i++) {
+            if (fabs(compareVec[i] - finalVec[i]) > 0.01) {

```

Comment:
1. Please use scientific notation for the values ​​in the res file.    
2. Regarding precision differences, use the minimum difference in scientific notation or the equal sign.